src/server/pkg/work/work.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package work;
+option go_package = "github.com/pachyderm/pachyderm/src/server/pkg/work";
+
+// import "google/protobuf/any.proto";
+
+enum State {
+  RUNNING = 0;
+  COMPLETE = 1;
+  FAILED = 2;
+}
+
+message Work {
```

Comment:
How opposed would you be to calling this a `Task`? I think that's a much more standard name for a unit of work. I think it would make some comments clearer too, there are comments above that refer to a "unit of work," and it's not entirely clear to me that "unit of work" is a synonym for a `Work`. I'm pretty sure it is, but if it was called a `Task` then you could just use the term as is in the comments. I.e. it could say: "Run runs the master with a given context and task." And there wouldn't be any ambiguity.

Reply:
Not opposed at all. I think it makes sense. I was sort of thinking of `work` as kind of a place holder for a better name. I was thinking about `Task` as well, especially as I was writing the documentation, but I left it as is to see what name might be suggested. What do you think the package name should be with this change? Also, does shard seem like a reasonable name for a piece of a Task? A shard would be a path range in the distributed merge case and a datum set in the job processing case.

Reply:
I think `work` is still a reasonable name for the package, we could call it `task` but then you have `task.Task` which stutters. With `work` we then have `work.Worker` which feels pretty reasonable to me. `Shard` also seems like a good name to me, the other option in my mind is `Chunk` but I think we use chunk in so many places it's a worse choice. And I think most programmers think of a distributed piece of work when they see `shard`.

Reply:
Sounds good.

Reply:
I thought about this a little bit more, and I think it might make sense to change `Shard` to `Task` instead. I think the way to look at it is that you can setup work that is split among multiple tasks. `Task` does not really fit the primary data structure as I am envisioning it now because the complete work to be done is represented by this data structure. `Task` could serve as a replacement name for `Shard` though and might make more sense that way. The client would be responsible for specifying tasks (it would understand that tasks for its use case are created by sharding a path range, set of datums, etc.) and the work package would just simply view them as tasks that workers need to do.

Reply:
So in this naming scheme a `Task` would be something that a single worker works on, rather than the broader unit of work. The name `Work` for the broader data structure still seems a little bit weird to me. I think `Task` actually desribes both of these things pretty well, and it might make sense to call them `Task` and `Subtask`, a task being made up of one or more subtasks seems pretty intuitive to me.

Reply:
I can do `Task` and `Subtask` if you really want to use those terms.

Reply:
Talked about offline, but I think having it be a recursive struct called `Task` that contains itself is probably the best option.